src/python/pants/backend/python/framework/stevedore/rules.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from collections import defaultdict
+
+from pants.backend.python.framework.stevedore.target_types import (
+    StevedoreExtensionTargets,
+    StevedoreNamespacesField,
+    StevedoreNamespacesProviderTargetsRequest,
+)
+from pants.backend.python.goals.pytest_runner import PytestPluginSetup, PytestPluginSetupRequest
+from pants.backend.python.target_types import (
+    PythonDistribution,
+    PythonDistributionEntryPoint,
+    PythonDistributionEntryPointsField,
+    ResolvedPythonDistributionEntryPoints,
+    ResolvePythonDistributionEntryPointsRequest,
+)
+from pants.engine.fs import EMPTY_DIGEST, CreateDigest, Digest, FileContent, PathGlobs, Paths
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+
+
+class GenerateEntryPointsTxtFromStevedoreExtensionRequest(PytestPluginSetupRequest):
+    @classmethod
+    def is_applicable(cls, target: Target) -> bool:
+        # select python_tests targets with stevedore_namespaces field
+        return (
+            target.has_field(StevedoreNamespacesField)
+            and target.get(StevedoreNamespacesField).value is not None
+        )
+
+
+@rule(
+    desc=f"Generate entry_points.txt to imitate `{PythonDistribution.alias}` installation.",
+    level=LogLevel.DEBUG,
+)
+async def generate_entry_points_txt_from_stevedore_extension(
+    request: GenerateEntryPointsTxtFromStevedoreExtensionRequest,
+) -> PytestPluginSetup:
+    requested_namespaces = request.target[StevedoreNamespacesField]
+    if not requested_namespaces.value:
+        return PytestPluginSetup(EMPTY_DIGEST)
+
+    stevedore_targets = await Get(
+        StevedoreExtensionTargets,
+        StevedoreNamespacesProviderTargetsRequest(requested_namespaces),
+    )
```

Comment:
Future note: If we implement the rule described in these comments (add `entry_points.txt` for all entry points in a `python_distribution` if the distribution is a dep of the `python_test`):
- https://github.com/pantsbuild/pants/issues/15481#issuecomment-1412936223
- https://github.com/pantsbuild/pants/issues/11386#issuecomment-1413032143

Then, here, we should probably skip any `python_distribution` targets that are already direct or transitive deps of the requested target. That way the other, more complete, `entry_points.txt` will not get clobbered by the partial one generated here.